test_scripts/Defects/8_0/3788/common_3788.lua

```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------------
+-- Common module for tests of https://github.com/smartdevicelink/sdl_core/issues/3788 issue
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local rc = require('user_modules/sequences/remote_control')
+local runner = require('user_modules/script_runner')
+local utils = require('user_modules/utils')
+
+--[[ General configuration parameters ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 2
+
+--[[ Common Variables ]]
+local common = { }
+
+--[[ Common Functions ]]
+common.Title = runner.Title
+common.Step = runner.Step
+common.preconditions = actions.preconditions
+common.start = rc.rc.start
+common.registerApp = actions.app.register
+common.registerAppWOPTU = actions.app.registerNoPTU
+common.activateApp = actions.app.activate
+common.postconditions = actions.postconditions
+common.policyTableUpdate = actions.policyTableUpdate
+common.EMPTY_ARRAY = utils.json.EMPTY_ARRAY
+common.getParams = actions.app.getParams
+common.setHMIAppId = actions.app.setHMIId
+common.createSession = actions.mobile.createSession
+common.getHMIConnection = actions.getHMIConnection
+common.unregisterApp = actions.app.unRegister
+
+local function getRCAppConfig(pPt)
+  local moduleTypes = rc.data.getRcModuleTypes()
+  local groups = { "Base-4", "RemoteControl" }
+  local appHMIType = { "REMOTE_CONTROL" }
+  if pPt then
+    local out = utils.cloneTable(pPt.policy_table.app_policies.default)
+    out.moduleType = moduleTypes
+    out.groups = groups
+    out.AppHMIType = appHMIType
+    return out
+  else
+    return {
+      keep_context = false,
+      steal_focus = false,
+      priority = "NONE",
+      default_hmi = "NONE",
+      moduleType = moduleTypes,
+      groups = groups,
+      AppHMIType = appHMIType
+    }
+  end
+end
+
+function common.preparePreloadedPT(pCountOfRCApps)
+  if not pCountOfRCApps then pCountOfRCApps = 1 end
+  local preloadedTable = actions.sdl.getPreloadedPT()
+  for i = 1, pCountOfRCApps do
+    local appId = actions.app.getParams(i).fullAppID
+    preloadedTable.policy_table.app_policies[appId] = getRCAppConfig(preloadedTable)
+  end
+  preloadedTable.policy_table.functional_groupings["DataConsent-2"].rpcs = utils.json.null
+  actions.sdl.setPreloadedPT(preloadedTable)
+end
+
+function common.rpcDenied(pModuleType, pAppId, pRpc, pResultCode)
+  local moduleData = rc.predefined.getSettableModuleControlData(pModuleType)
+  rc.rc.rpcReject(pModuleType, moduleData.moduleId, pAppId, pRpc, moduleData, pResultCode)
+end
+
+function common.rpcAllowed(pModuleType, pAppId, pRpc)
+  local moduleData = rc.predefined.getSettableModuleControlData(pModuleType)
+  rc.rc.rpcSuccess(pModuleType, moduleData.moduleId, pAppId, pRpc, moduleData, false)
+end
+
+return common
```

Comment:
I would propose to get rid of common module since there is no a lot of specific logic in it.
We usually do not create common modules for the issues unless a lot of generic logic is required.
It looks like we would have aprox. the same amount of code (comparing total number of lines with and without common)